### PR TITLE
Companion map

### PR DESCRIPTION
**Pull request by pjaudiomv**
this adds in companion map using new show_map attribute

ref #52 

example
[https://sca.charlestonna.org/test-tab-map/](https://sca.charlestonna.org/test-tab-map/)

### CODE REVIEW

File: js/bmlt_tabs_map.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
+
+	var map;
```
[the remark this answers is not in the document]
Comment (pjaudiomv) [reply js/bmlt_tabs_map.js]: all of the examples on google have it being set that way
https://developers.google.com/maps/documentation/javascript/examples/map-simple
I was just following along, but yea doesn't make sense as it's not needed globally anywhere

File: readme.txt [hunk 2]
```diff
@@ -38,6 +38,7 @@ Crouton is a "Fork" of the BMLT Tabbed UI plugin. This plugin provides a Tabbed
 
 = 2.3.3 = *UNRELEASED*
 * added case-insensitive sorting.
```
[the remark this answers is not in the document]
Comment (pjaudiomv) [reply readme.txt]: done

File: js/bmlt_tabs_map.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+		oms.addListener('click', function(marker) {
+			infoWindow.setContent(marker.desc);
+			infoWindow.open(map, marker);
+		});
+		// Add some markers to the map.
+		// Note: The code uses the JavaScript Array.prototype.map() method to
+		// create an array of markers based on a given "locations" array.
+		// The map() method here has nothing to do with the Google Maps API.
+		var markers = meetingData.map(function(location, i) {
+			var weekdays = ['ERROR', 'Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
```
[the remark this answers is not in the document]
Comment (pjaudiomv) [reply js/bmlt_tabs_map.js]: done

File: crouton.php [hunk 4]
```diff
@@ -1051,6 +1067,16 @@ public function adminOptionsPage()
                             </li>
                         </ul>
                     </div>
+                    <div style="padding: 0 15px;" class="postbox">
+                        <h3>Google Api Key</h3>
+                        <p>This is only needed when using the companion map feature show_map.</p>
```
[the remark this answers is not in the document]
Comment (pjaudiomv) [reply crouton.php]: done

File: crouton.php [hunk 5]
```diff
@@ -1051,6 +1067,16 @@ public function adminOptionsPage()
                             </li>
                         </ul>
                     </div>
+                    <div style="padding: 0 15px;" class="postbox">
+                        <h3>Google Api Key</h3>
```
[the remark this answers is not in the document]
Comment (pjaudiomv) [reply crouton.php]: done

File: crouton.php [hunk 6]
```diff
@@ -913,6 +928,7 @@ public function adminOptionsPage()
                 $this->options['recurse_service_bodies'] = $_POST['recurse_service_bodies'];
                 $this->options['extra_meetings'] = $_POST['extra_meetings'];
                 $this->options['extra_meetings_enabled'] = intval($_POST['extra_meetings_enabled']);
+                $this->options['google_api_key']   = $_POST['google_api_key'];
```
[the remark this answers is not in the document]
Comment (pjaudiomv) [reply crouton.php]: done

File: crouton.php [hunk 7]
```diff
@@ -306,11 +306,21 @@ public function tabbedUi($atts, $content = null)
...
+                "used_formats" => '0',
+                "show_map" => '0'
             ), $atts));
             if ($show_distance == '1') {
                 wp_enqueue_script("bmlt-tabs-distance", plugin_dir_url(__FILE__) . "js/bmlt_tabs_distance.js", array('jquery'), filemtime(plugin_dir_path(__FILE__) . "js/bmlt_tabs_distance.js"), true);
             }
+            if ($show_map == '1') {
+                wp_enqueue_script("markerclusterer", plugin_dir_url(__FILE__) . "js/markerclusterer.js", array('jquery'), filemtime(plugin_dir_path(__FILE__) . "js/markerclusterer.js"), true);
+                wp_enqueue_script("oms", plugin_dir_url(__FILE__) . "js/oms.min.js", array('jquery'), filemtime(plugin_dir_path(__FILE__) . "js/oms.min.js"), true);
+                wp_enqueue_script('google-maps', 'https://maps.googleapis.com/maps/api/js?key=' . $this->options['google_api_key'], '', '');
```
[the remark this answers is not in the document]
Comment (pjaudiomv) [reply crouton.php]: done

File: js/bmlt_tabs_map.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (pjaudiomv) [reply js/bmlt_tabs_map.js]: I removed this as well

File: readme.txt [hunk 8]
```diff
@@ -37,7 +37,8 @@ Crouton is a "Fork" of the BMLT Tabbed UI plugin. This plugin provides a Tabbed
 == Changelog ==
 
 = 2.3.3 = *UNRELEASED*
```
Comment (radius314) [inline readme.txt]: I would call this 2.4.0

File: crouton.php [hunk 9]
```diff
@@ -340,6 +350,9 @@ public function tabbedUi($atts, $content = null)
             if ($include_weekday_button != '0' && $include_weekday_button != '1') {
                 return '<p>crouton Error: include_weekday_button must = "0" or "1".</p>';
             }
+            if ($show_map == '1' && $this->options['google_api_key'] == '') {
```
[the remark this answers is not in the document]
Comment (pjaudiomv) [reply crouton.php]: makes sense, done.

File: crouton.php [hunk 10]
```diff
@@ -306,11 +306,26 @@ public function tabbedUi($atts, $content = null)
...
+                wp_enqueue_script("markerclusterer", plugin_dir_url(__FILE__) . "js/markerclusterer.js", array('jquery'), filemtime(plugin_dir_path(__FILE__) . "js/markerclusterer.js"), true);
+                wp_enqueue_script("oms", plugin_dir_url(__FILE__) . "js/oms.min.js", array('jquery'), filemtime(plugin_dir_path(__FILE__) . "js/oms.min.js"), true);
+                wp_enqueue_script('google-maps', 'https://maps.googleapis.com/maps/api/js?key=' . $this->options['google_api_key'], '', '');
+                wp_enqueue_script("bmlt-tabs-map", plugin_dir_url(__FILE__) . "js/bmlt_tabs_map.js", array('jquery'), filemtime(plugin_dir_path(__FILE__) . "js/bmlt_tabs_map.js"), true);
+                wp_localize_script('bmlt-tabs-map', 'bmltTabsMap', array(
+                    'pluginUrl' => plugin_dir_url(__FILE__),
+                    'maxZoomLevel' => $max_zoom_level,
+                ));
+            }
+            if ($show_map == '1' && $this->options['google_api_key'] == '') {
```
Comment (radius314) [inline crouton.php]: these two if statements can be combined in a sense.

Comment (pjaudiomv) [reply crouton.php]: not sure if thats what you meant but in a sense i combined them ha

**Pull request merged**